src/main/scala/org/apache/spark/ml/feature/RapidsPCA.scala

```diff
@@ -168,26 +185,61 @@ class RapidsPCAModel(
    */
   override def transform(dataset: Dataset[_]): DataFrame = {
     val outputSchema = transformSchema(dataset.schema, logging = true)
+    val cols_A = dataset.select($(transformInputCol)).first.get(0)
+      .asInstanceOf[mutable.WrappedArray[Double]].size
+    val gpuIdBC = dataset.rdd.context.broadcast(getGpuId)
 
-    // TODO(rongou): make this faster and re-enable.
-    //    if (getUseGemm) {
-    //      val transformed = dataset.toDF().rdd.mapPartitions(iterator => {
-    //        val gpuID = TaskContext.get().resources()("gpu").addresses(0)
-    //        val partition = iterator.toList
-    //        val A = Matrices.fromVectors(partition.map(_.getAs[Vector]($(inputCol)))).toDense
-    //        val C = Matrices.zeros(partition.length, getK).toDense
-    //        CUBLAS.gemm_b(A, pc, C, gpuID)
-    //        C.rowIter.zip(partition.iterator).map { case (v, r) =>
-    //          Row.fromSeq(r.toSeq ++ Seq(v))
-    //        }
-    //      })
-    //      dataset.sparkSession.createDataFrame(transformed, outputSchema)
-    //    }
-    //    else {
-    val transposed = pc.transpose
-    val transformer = udf { vector: Vector => transposed.multiply(vector) }
-    dataset.withColumn($(outputCol), transformer(col($(inputCol))), outputSchema($(outputCol)).metadata)
-    //    }
+    /**
+     * UDF class to speedup transform process of PCA
+     */
+    class gpuTransform extends Function[mutable.WrappedArray[Double], Array[Double]]
+      with RapidsUDF with Serializable {
+      override def evaluateColumnar(args: ColumnVector*): ColumnVector = {
+        logDebug("==========using GPU transform==========")
+        val gpu = if (gpuIdBC.value == -1) {
+          TaskContext.get().resources()("gpu").addresses(0).toInt
+        } else {
+          gpuIdBC.value
+        }
+        require(args.length == 1, s"Unexpected argument count: ${args.length}")
+        val input = args.head
+        val rows_A = input.getRowCount.toInt
+        val childCViewNative = input.getChildColumnView(0).getNativeView
```

Comment:
`getChildColumnView` returns a native resource in host memory and is leaked here since it is never closed.  This needs to track the `ColumnView` returned in a try..finally block to ensure it is always closed.

Reply:
Updated, now pass `input`(parent ColumnView) to native directly.